src/services/prices/price-sources/aggregator-price-source.ts

```diff
@@ -80,6 +80,45 @@ export class AggregatorPriceSource implements IPriceSource {
     return this.aggregate(collected, aggregateBulkHistoricalPrices);
   }
 
+  async getChart({
+    tokens,
+    span,
+    period,
+    bound,
+    searchWidth,
+    config,
+  }: {
+    tokens: Record<ChainId, TokenAddress[]>;
+    span: number;
+    period: TimeString;
+    bound: { from: Timestamp } | { upTo: Timestamp | 'now' };
+    searchWidth?: TimeString;
+    config: { timeout?: TimeString } | undefined;
+  }): Promise<Record<ChainId, Record<TokenAddress, PriceResult[]>>> {
+    const collected = await collectAllResults({
+      allSources: this.sources,
+      fullRequest: tokens,
+      query: 'getChart',
+      getResult: (source, filteredRequest, sourceTimeout) =>
+        source.getChart({
+          tokens: filteredRequest,
+          span,
+          period,
+          bound,
+          searchWidth,
+          config: { timeout: sourceTimeout },
+        }),
+      timeout: config?.timeout,
+    });
+    // TODO: Support more than one source
+    return Object.fromEntries(
+      Object.entries(collected).map(([chain, tokens]) => [
+        parseInt(chain),
+        Object.fromEntries(Object.entries(tokens).map(([token, prices]) => [token, prices[0]])),
+      ])
+    );
```

Comment:
May I suggest something a little bit simpler, like this?
```suggestion
    // TODO: Support more than one source
    const sourcesInChains = getSourcesThatSupportRequestOrFail(tokens, this.sources, 'getChart');
    return sourcesInChains[0].getChart({ tokens, span, period, bound, searchWidth, config }) ?? {}
```

Reply:
Making things complicated is my middle name